plenum/common/ledger_manager.py

```diff
@@ -17,17 +17,40 @@
     ConsistencyProof, f, CatchupReq
 from plenum.common.metrics_collector import MetricsCollector, NullMetricsCollector, measure_time, MetricsName
 from plenum.common.util import compare_3PC_keys, SortedDict, min_3PC_key
+from plenum.server.catchup.utils import CatchupDataProvider
 from plenum.server.has_action_queue import HasActionQueue
 from plenum.server.quorums import Quorums
 from stp_core.common.log import getlogger
 
 logger = getlogger()
 
 
+class CatchupNodeDataProvider(CatchupDataProvider):
```

Comment:
I think we will need to add `transform_txn_for_ledger` here

Reply:
We'll see when it comes to implementing `NodeLeecherService`, I'm just trying to keep this interface minimal and add things only when they are really needed.